<launch>

	<arg name="port" default="5000" />
	<arg name="port_rosbridge" default="5001" />
	<arg name="base_url" default="" /> <!-- e.g. /vizanti -->

	<arg name="retry_startup_delay" default="10" />
	<arg name="fragment_timeout" default="30" />
	<arg name="delay_between_messages" default="0" />
	<arg name="max_message_size" default="None" />
	<arg name="flask_debug" default="True" />

	<!-- Launch rosbridge without https so it works in AP mode. -->
	<node name="vizanti_rosbridge" pkg="rosbridge_server" type="rosbridge_websocket" output="screen">
		<param name="authenticate" value="false" />
		<param name="port" value="$(arg port_rosbridge)"/>
		<param name="address" value=""/>
		<param name="retry_startup_delay" value="$(arg retry_startup_delay)"/>
		<param name="fragment_timeout" value="$(arg fragment_timeout)"/>
		<param name="delay_between_messages" value="$(arg delay_between_messages)"/>
		<param name="max_message_size" value="$(arg max_message_size)"/>
		<param name="unregister_timeout" value="99999999"/> <!-- https://github.com/RobotWebTools/rosbridge_suite/issues/298 -->
	</node>

	<node name="vizanti_rosapi" pkg="rosapi" type="rosapi_node" />

	<node name="vizanti_flask_node" pkg="vizanti" type="server.py" output="screen">
		<param name="host" value="0.0.0.0"/>
		<param name="port" value="$(arg port)"/>
		<param name="port_rosbridge" value="$(arg port_rosbridge)"/>
		<param name="base_url" value="$(arg base_url)"/>
		<param name="flask_debug" value="$(arg flask_debug)"/>
	</node>

	<node name="vizanti_topic_handler_node" pkg="vizanti" type="topic_handler.py" output="screen"/>
	<node name="vizanti_service_handler_node" pkg="vizanti" type="service_handler.py" output="screen"/>
    
</launch>
